### DISCUSSION TITLE
Problem Deleting All Text

### DISCUSSION THREAD

**Discussion by hughgoodbody** [mozman/ezdxf#804] [Q&A]
I am having a problem deleting all the text entities in a dxf. The code I am using is:

        #### READ DXF FILE ####
    dwg = ezdxf.readfile(dxfFile)
    #print(dwg)
    msp = dwg.modelspace()
    #Delete existing
    myquery = msp.query('TEXT')
    print(myquery)
    for entity in myquery:
      msp.delete_entity(entity)

The text objet is found, but not deleted. Am I missing something?

**Comment by mozman** [comment]
Check this `assert len(msp.query('TEXT')) == 0` at the end of your script.

If you did the check with a CAD application, did you write back the DXF file?

**Reply by hughgoodbody** [reply]
Thanks, found the issue, a missing bracket!
I am also using the bin packing add-in and want to position some dxf blocks where the bin pack blocks are positioned, is there a way that the bin packing add in returns the position or centre points of the blocks it creates?

**Reply by mozman** [reply]
The `Item` class has a `bbox` property which returns a `ezdxf.math.BoundingBox` object:

```Python
for item in packer.bins[0]:
    print(item.bbox.center)
```
The origin (0, 0, 0) is the "lower-left" corner of the container.

**Reply by hughgoodbody** [reply]
Perfect, thanks so much, my last question, is there a way to stop items which are being packed from being rotated? I have been trying to figure out the RotationType and how to use it, but am a bit stuck!
